### PR TITLE
Improvements to e2e logging

### PR DESCRIPTION
**Pull request by james-chf**
Relates to anoma/namada#150

- For any command we run during e2e tests, log all output to a file in the temporary directory
- explicitly mention we send `Ctrl+C` to commands rather than just passively waiting for them to finish
- ensure `AnomaCmd` expect failures log the command that the failure happened for
- record timings for all tests (closes https://github.com/anoma/namada/issues/147, from https://github.com/anoma/anoma/pull/1188)

If a failure happens, log command
```
Error:
   0: Reached a timeout for expect type of command
      Command: /opt/workspace/heliax/repos/anoma/abcipp/target/debug/anomac --base-dir /var/folders/cy/fxxsxwd56yg67d68vxj6l52h0000gn/T/.tmpq7k0N2 --mode full transfer --source Bertha --target Albert --token XAN --amount 10.1 --fee-amount 0 --gas-limit 0 --fee-token XAN --ledger-address 127.0.0.1:28657
       Needle: Transaction applied

Location:
   tests/src/e2e/setup.rs:487

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                                ⋮ 10 frames hidden ⋮
  11: anoma_tests::e2e::setup::AnomaCmd::exp_string::{{closure}}::h7a448153db6d5ac3
      at /opt/workspace/heliax/repos/anoma/abcipp/tests/src/e2e/setup.rs:487
  12: core::result::Result<T,E>::map_err::h173d24cf5428418f
      at /rustc/cacc75c82ebe15cf63d31034fcf7f1016cddf0e2/library/core/src/result.rs:855
  13: anoma_tests::e2e::setup::AnomaCmd::exp_string::h12e21a8761bc416f
      at /opt/workspace/heliax/repos/anoma/abcipp/tests/src/e2e/setup.rs:486
  14: anoma_tests::e2e::ledger_tests::ledger_many_txs_in_a_block::{{closure}}::{{closure}}::h974d35ab3a39a1b6
      at /opt/workspace/heliax/repos/anoma/abcipp/tests/src/e2e/ledger_tests.rs:999
                                ⋮ 13 frames hidden ⋮
```

In any case, log the command being run
```
> Running:
/opt/workspace/heliax/repos/anoma/abcipp/target/debug/anomac --base-dir /var/folders/cy/fxxsxwd56yg67d68vxj6l52h0000gn/T/.tmpq7k0N2 --mode full transfer --source Bertha --target Albert --token XAN --amount 10.1 --fee-amount 0 --gas-limit 0 --fee-token XAN --ledger-address 127.0.0.1:28657
```

### CODE REVIEW

**Comment by Fraccaman** [comment]
we have `ANOMA_LOG_COLOR` env variable to disable/enable ANSI sequenes. Maybe it can help.

**Comment by james-chf** [comment]
I am still trying to figure out why `e2e::ledger_tests::ledger_many_txs_in_a_block` is failing for ABCI++. It does fail locally for me. The same test does pass in `v0.6.1` with ABCI++ so this PR has changed something to make it start failing. 

I've tried increasing the consensus timeout and reducing the number of spawned `anomac tx` commands from 3 to 1, but the test still fails with errors like:

```
Error:
   0: Reached a timeout for expect type of command
      Command: /opt/workspace/heliax/repos/anoma/abcipp/target/debug/anomac --base-dir /var/folders/cy/fxxsxwd56yg67d68vxj6l52h0000gn/T/.tmpiLotIB --mode full transfer --source Bertha --target Albert --token XAN --amount 10.1 --fee-amount 0 --gas-limit 0 --fee-token XAN --ledger-address 127.0.0.1:28657
       Needle: Transaction accepted
```

Interestingly this error happens even though the timeout for that command is 40s and the test completes in less than 40s.

**Comment by james-chf** [comment]
This PR can pass CI now, the test failure was happening as when calling `session.with_log` in expectrl, a new instance was returned without the existing session configuration copied over e.g. max timeout. So the test was failing due to too short of a timeout.

**Comment by james-chf** [comment]
going to make this PR agnostic to expectrl so that it can be used even while we revert to rexpect in https://github.com/anoma/anoma/pull/1219

**Comment by james-chf** [comment]
pls update wasm

**Comment by james-chf** [comment]
changed back to log with expectrl

File: tests/src/e2e/setup.rs [hunk 1]
```diff
@@ -683,9 +714,37 @@ where
             e
         )
     })?;
+
+    let log_path = {
+        let mut rng = rand::thread_rng();
+        let log_dir = base_dir.as_ref().join("logs");
```
[the remark this answers is not in the document]
Comment (tzemanovic) [reply tests/src/e2e/setup.rs:720]: ah, yeh, it's relative to the `base_dir` which is nested for validators. We could pass in another arg to `run_cmd` from `Test::run_cmd_as` that would be just `self.base_dir` of the `Test` itself

**Review by tzemanovic — APPROVED** [review]
LGTM, thanks!

**Comment by tzemanovic** [comment]
merged via https://github.com/anoma/namada/pull/222

**Pull request closed without merging**